src/pip/_internal/pep425tags.py

```diff
@@ -104,6 +104,26 @@ def _get_custom_interpreter(implementation=None, version=None):
     return "{}{}".format(implementation, version)
 
 
+def _with_legacy_tags(standard_tags, interpreter, impl):
+    # type: (List[Tag], str, Optional[str]) -> List[Tag]
+    """For backwards compatibilty, add legacy tags that pip used to accept"""
+    all_supported_tags = standard_tags  # Default to not making any changes
```

Comment:
nit: to avoid potential action-at-a-distance bugs, create a copy here?

```suggestion
    all_supported_tags = standard_tags[:]  # Default to not making any changes
```

Reply:
This is an internal helper that is going to get called quite a bit, so I expect doing that extra work every time would pointlessly waste a non-trivial number of CPU cycles.

Instead, I added a note to the docstring highlighting the fact that the given list may be returned directly, rather than the result necessarily being a new list.